server/aesdsocket.c

```diff
@@ -0,0 +1,423 @@
+        }
+
+        soc_client = accept(soc_server, (struct sockaddr*)&aesdsoc_addr, 
+            (socklen_t*)&aesdsoc_addr_len);
+        if (soc_client < 0 ) {
+            rc = -1;
+            syslog(LOG_ERR, "aesdsocket: API accept failed %s", strerror(errno));
+            goto exit_shutdown;
+        }
+
+        syslog(LOG_INFO, "Accepted connection from %s", inet_ntoa(aesdsoc_addr.sin_addr)); 
+
+                
+        rc = fcntl(soc_client, F_SETFL, O_NONBLOCK);
+        if (rc < 0) {
+            syslog(LOG_ERR, "aesdsocket: fcntl failed %s", strerror(errno));
+            goto exit_client;
+        }
+        rc = clock_gettime(CLOCK_MONOTONIC, &start);
+        if (rc < 0) {
+            syslog(LOG_ERR, "aesdsocket: clock_gettime failed %s", strerror(errno));
+            goto exit_client;
+        }
+
+        int total_bytes_received = 0;
+        buffer = (char*)malloc(buffer_size);
+        if(buffer == NULL) {    
+            rc = -1;
+            syslog(LOG_ERR, "aesdsocket: malloc failed %s", strerror(errno));
+            goto exit_client;
+        }
+        while(1) {
+            rcv_data_len = recv(soc_client, buffer + total_bytes_received, buffer_size - total_bytes_received, 0);
+            if (rcv_data_len < 0) {
+                rc = clock_gettime(CLOCK_MONOTONIC, &now);
+                if (rc < 0) {            
+                    syslog(LOG_ERR, "aesdsocket: clock_gettime failed %s", strerror(errno));
+                    goto goto_mem_cleanup;
+                }
+                diff_time = ((now.tv_sec - start.tv_sec) * 1000000) + ((now.tv_nsec - start.tv_nsec) / 1000); 
+                if(diff_time > MSEC_2_USEC(PACKET_TIMEOUT_END)) {   
+                    file_buffer = (char*)malloc(total_bytes_received+1);
+                    if(file_buffer == NULL) {
+                        rc = -1;
+                        if(buffer) free(buffer);
+                        syslog(LOG_ERR, "aesdsocket: malloc failed %s", strerror(errno));
+                        goto goto_mem_cleanup;
+                    }
+                    if (process_and_save_data(buffer, file_buffer, 
+                            total_bytes_received, &wr_pointer, fp) < 0) {
+                        rc = -1;
+                        syslog(LOG_ERR, "aesdsocket: process_and_save_data");
+                        goto goto_mem_cleanup;
+                    }
+                    int file_len = ftell(fp);
+                    /* 
+                    ** Timeout reached and if not data is saved, discard the packet.
+                    ** If file consists of data, send /replay the buffer and then wait for 
+                    ** new connection.
+                    */
+                    if (file_len == 0) {
+                        syslog(LOG_ERR, "aesdsocket: No data saved, after 100 ms timeout rejecting packet");
+                        goto discard_packet;
+                    }
+                    tx_buf = (char *) malloc(file_len);
```

Comment:
@sujoyray-ucolorado what assumption are you making about the size of the process heap versus the size of all packets sent here?